### PR TITLE
Add support for Gitea pull requests

### PR DESCRIPTION
**Pull request by qwerty287**
## Proposed changes

Add support for Gitea pull requests, similar to GitHub and GitLab. Gitea's API is very similar to GitHub, implementation is similar.

Closes https://github.com/WeblateOrg/weblate/issues/7099

## Checklist

- [ ] Lint and unit tests pass locally with my changes. **(see note below)**
- [x] I have added tests that prove my fix is effective or that my feature works.
- [x] I have added documentation to describe my feature.
- [x] I have squashed my commits into logic units.
- [x] I have described the changes in the commit messages.

## Other information

Unfortunately, I wasn't able to test my changes. I can't setup a development environment, even if I'm following the instructions in your docs. (Docker dev image isn't working as well - I don't know why) Maybe this is incomplete and not working, but Gitea is similar to GitHub so most code can be copied. I'm really sorry for that, but I don't know how to fix these issues. I hope I got everything in to make it working.

Gitea allows pull requests without commits, so in this case there is no error how it is handled on GitHub, instead an empty PR is created.

### CODE REVIEW

File: docs/admin/config.rst [hunk 1]
```diff
@@ -725,6 +725,63 @@ GET_HELP_URL
 
 URL where support for your Weblate instance can be found.
 
+.. setting:: GITEA_CREDENTIALS
+
+GITEA_CREDENTIALS
+------------------
+
+.. versionadded:: 4.3
```
Comment (qwerty287) [inline docs/admin/config.rst:733]: This is just copied right now, what should I add as the next version number?

File: dev-docker/docker-compose.yml [hunk 2]
```diff
@@ -33,6 +33,10 @@ services:
     environment:
       DJANGO_SETTINGS_MODULE: weblate.settings_docker
       WEBLATE_SITE_DOMAIN: ${WEBLATE_HOST}
+      WEBLATE_GITHUB_USERNAME: qwerty287
+      WEBLATE_GITHUB_TOKEN: 95a2a068275ced3452db6d57aed5fa0ade82a50f
+      WEBLATE_GITEA_USERNAME: qwerty287
+      WEBLATE_GITEA_TOKEN: 95a2a068275ced3452db6d57aed5fa0ade82a50f
```
Comment (nijel) [inline dev-docker/docker-compose.yml:39]: If you need some changes, add them to the override file, not here.

File: docs/admin/config.rst (hunk 1 above)
Comment (nijel) [reply docs/admin/config.rst:733]: Please use 4.12 here.

File: docs/admin/config.rst [hunk 3]
```diff
@@ -725,6 +725,63 @@ GET_HELP_URL
...
+    GITHUB_CREDENTIALS = {
+        "api.github.com": {
+            "username": "weblate",
+            "token": "your-api-token",
+        },
+        "github.example.com": {
+            "username": "weblate",
+            "token": "another-api-token",
+        },
+    }
```
Comment (nijel) [inline docs/admin/config.rst:753]: The example is for GitHub, not for Gitea...

File: docs/admin/config.rst [hunk 4]
```diff
@@ -725,6 +725,63 @@ GET_HELP_URL
...
+
+.. setting:: GITEA_USERNAME
+
+GITEA_USERNAME
+---------------
+
```
Comment (nijel) [inline docs/admin/config.rst:759]: Version added is missing here.

File: weblate/vcs/tests/test_vcs.py [hunk 5]
```diff
@@ -501,6 +507,13 @@ def add_remote_commit(self, conflict=False, rename=False):
             self._class = backup
 
 
+@override_settings(GITLAB_USERNAME="test")
```
Comment (nijel) [inline weblate/vcs/tests/test_vcs.py:510]: Should configure Gitea, not GitLab.

File: weblate/vcs/tests/test_vcs.py [hunk 6]
```diff
@@ -501,6 +507,13 @@ def add_remote_commit(self, conflict=False, rename=False):
...
+class VCSGiteaTest(VCSGitTest):
+    _class = GiteaFakeRepository
+    _vcs = "git"
+    _sets_push = False
+
```
Comment (nijel) [inline weblate/vcs/tests/test_vcs.py:515]: Responses mocking is missing here to actually test the integration.

File: docs/admin/config.rst [hunk 7]
```diff
@@ -725,6 +725,63 @@ GET_HELP_URL
...
+.. versionadded:: 4.3
+
+Gitea personal access token used to make API calls to send pull requests for
+translation updates.
+
+.. seealso::
+
+   :setting:`GITEA_CREDENTIALS`,
+   :ref:`vcs-gitea`,
```
Comment (nijel) [inline docs/admin/config.rst:780]: The target for this is missing, the documentation for Gitea should be in `docs/vcs.rst`.

**Review by nijel — COMMENTED** [review]
What kind of issues do you have with the devel Docker? Please open a separate issue for that.

File: dev-docker/docker-compose.yml (hunk 2 above)
Comment (qwerty287) [reply dev-docker/docker-compose.yml]: Sorry, this was just for testing purposes and I forgot to remove it.

File: docs/admin/config.rst [hunk 8]
```diff
@@ -725,6 +725,65 @@ GET_HELP_URL
 
 URL where support for your Weblate instance can be found.
 
+.. setting:: GITEA_CREDENTIALS
+
+GITEA_CREDENTIALS
+------------------
```
Comment (nijel) [inline docs/admin/config.rst:731]:
```suggestion
GITEA_CREDENTIALS
-----------------
```

File: docs/admin/config.rst [hunk 9]
```diff
@@ -725,6 +725,65 @@ GET_HELP_URL
...
+        "gitea.example.com": {
+            "username": "weblate",
+            "token": "another-api-token",
+        },
+    }
+
+.. setting:: GITEA_USERNAME
+
+GITEA_USERNAME
+---------------
```
Comment (nijel) [inline docs/admin/config.rst:758]:
```suggestion
GITEA_USERNAME
--------------
```

File: docs/admin/config.rst [hunk 10]
```diff
@@ -725,6 +725,65 @@ GET_HELP_URL
...
+
+.. seealso::
+
+   :setting:`GITEA_CREDENTIALS`,
+   :ref:`vcs-gitea`
+
+.. setting:: GITEA_TOKEN
+
+GITEA_TOKEN
+------------
```
Comment (nijel) [inline docs/admin/config.rst:772]:
```suggestion
GITEA_TOKEN
-----------
```

**Review by nijel — COMMENTED** [review]
Fixing documentation underlines to match title length.

File: weblate/locale/en_GB/LC_MESSAGES/django.po [hunk 11]
```diff
@@ -14886,6 +14886,10 @@ msgstr ""
 msgid "GitHub pull request"
 msgstr ""
 
+#: weblate/vcs/git.py:967
+msgid "Gitea pull request"
+msgstr ""
+
```
Comment (nijel) [inline weblate/locale/en_GB/LC_MESSAGES/django.po:14892]: Not needed:

```suggestion
```

**Comment by nijel** [comment]
Further fixes in https://github.com/qwerty287/weblate/pull/1

**Comment by nijel** [comment]
Rebased, squashed and fixed tests are in https://github.com/WeblateOrg/weblate/pull/7394

**Pull request closed without merging**